components/IndoorDirections/TypesOfDirections/DifferentBuildingDirections.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { Line, G } from 'react-native-svg';
+import { ClassGraph } from '../../../constants/ClassGraph';
+import { dijkstra, getFloorNumber, changeClassName } from '../Dijkstra/DijkstraAlgorithm';
+
+export function DifferentBuildingDirections(props) {
+    const rooms = props.rooms;
+    const graph = ClassGraph();
+    var goTo = props.from;
+    var tempGoTo = props.from;
+
+    if (props.from.includes(" ")) {
```

Comment:
Now I understand your line `const num = name.match(/\d+/g).map(Number);`  in the  _changeClassName_. However, in what case you will have a space in the string?

Reply:
In the case the from/to is a building, all our buildings have at least one " "

Reply:
hmmm I see, but if it's a building, it means, it's going to go on the _changeClassName_ ?  Also, you seem to be doing the same thing in the followin if statement but with _props.to_ so now I'm even more confused. By all means, your code works, I'm just trying to understand. 🔢

Reply:
Yes, it needs to go, imagine the scenario Grey Nuns to H535. H353 needs to be converted to H835

Reply:
To not drag this too much, can you just quickly explain to me what the first two if statements are attempting to do? 🥇

Reply:
For sure

`var goTo = props.from;`: this is the from classroom, imagine this being H513
`var tempGoTo = props.from;`: same thing as above.

If you go to line 53, we check the floor in which we need to print the path. Problem is, if I convert H513 to H813 too early, I would print the path on floor 8 which is not what we want. So, goTo is used to know which class to go to, tempGoTo is used to get the correct floor number

Reply:
I understand now! 👍